src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VsService`1.cs

```diff
@@ -22,10 +23,10 @@ internal class VsService<T> : IVsService<T>
         private readonly AsyncLazy<T> _value;
 
         [ImportingConstructor]
-        public VsService([Import(typeof(SAsyncServiceProvider))]IAsyncServiceProvider serviceProvider, IProjectThreadingService threadingService)
+        public VsService([Import(typeof(SAsyncServiceProvider))]IAsyncServiceProvider serviceProvider, JoinableTaskContext joinableTaskContext)
```

Comment:
Why not import the factory here instead of the context? All usages want a factory.

Reply:
Only the context is added to the MEF container.